server/src/test/java/org/elasticsearch/cluster/service/MasterServiceTests.java

```diff
@@ -263,9 +262,18 @@ public void testClusterStateTaskListenerThrowingExceptionIsOkay() throws Interru
         AtomicBoolean published = new AtomicBoolean();
 
         try (MasterService masterService = createMasterService(true)) {
+            ClusterStateTaskListener update = new ClusterStateTaskListener() {
+                @Override
+                public void clusterStateProcessed(ClusterState oldState, ClusterState newState) {
+                    throw new IllegalStateException();
```

Comment:
Got confused by this, thought it should perhaps be an assertion instead. This'd clarify:

```suggestion
                    throw new RuntimeException("testing exception handling");
```